test/unit/builtins/async-await.spec.ts

```diff
@@ -0,0 +1,242 @@
+import { ModuleKind, ScriptTarget } from "typescript";
+import { notAllowedTopLevelAwait } from "../../../src/transformation/utils/diagnostics";
+import * as util from "../../util";
+
+const promiseTestLib = `
```

Comment:
Can this be shared with the promises tests?

Reply:
We cannot export code from .spec.ts files, because next to the exported thing, when importing it all tests from the other file are also imported. Jest has a eslint rule that prevents this. I think just having the code on top of both files is clearer than putting it in a shared .ts file, even though it is a bit of duplication.